node/src/main/scala/org/bitcoins/node/SpvNode.scala

```diff
@@ -33,27 +67,105 @@ case class SpvNode(peer: Peer, chainApi: ChainApi)(
     PeerMessageSender(client, nodeAppConfig.network)
   }
 
-  /** Starts our spv node */
-  def start(): Future[SpvNode] = {
-    peerMsgSender.connect()
+  /**
+    * Sends the given P2P to our peer.
+    * This method is useful for playing around
+    * with P2P messages, therefore marked as
+    * `private[node]`.
+    */
+  private[node] def send(msg: NetworkPayload): Unit = {
+    peerMsgSender.sendMsg(msg)
+  }
 
-    val isInitializedF =
-      AsyncUtil.retryUntilSatisfied(peerMsgRecv.isInitialized)
+  /**
+    * Adds the given public key to our table of
+    * interesting public keys. This table is used
+    * to construct a bloom filter we send to our
+    * peer, which then will send us information
+    * about transactions related to said keys.
+    */
+  def addPubKey(pub: ECPublicKey): Future[SpvNode] = {
+    logger.debug(s"Marking $pub as an a public key of interest")
+    for {
+      _ <- pubkeyDAO.create(pub)
+    } yield this
+  }
 
-    isInitializedF.map { _ =>
-      logger.info(s"Our peer=${peer} has been initialized")
-    }
+  /**
+    * Starts our SPV node,
+    * initializing it with the correct DB setup
+    * and then connect to our peer
+    */
+  def start(): Future[SpvNode] = {
+    logger.info(s"Starting SPV node with datadir ${nodeAppConfig.datadir}")
+    for {
+      _ <- nodeAppConfig.initialize()
+      _ <- {
+        peerMsgSender.connect()
+
+        val isInitializedF =
+          AsyncUtil.retryUntilSatisfied(peerMsgRecv.isInitialized)
+
+        isInitializedF.onComplete {
+          case Failure(err) =>
+            logger.error(s"Failed to conenct with peer=$peer with err=${err}")
+          case Success(_) =>
+            logger.info(s"Our peer=${peer} has been initialized")
+        }
+
+        isInitializedF
+      }
+      node <- initBloomFilter()
+    } yield node
+  }
 
-    isInitializedF.failed.foreach { err =>
-      logger.error(s"Failed to conenct with peer=$peer with err=${err}")
+  /**
+    * Constructs and sends a bloom filter based on
+    * our table of pubkeys that's marked as interesting
+    *
+    * @return An SPV node with the newly set bloom filter
+    */
+  private def initBloomFilter(): Future[SpvNode] = {
```

Comment:
So I think I disagree with putting this here. I think the bloom filter should be constructed in the wallet project and available via the `WalletApi`. I think we should then be able to call something like `SpvNode.fromBloomFilter()` or something like that. I don't think it should be the SpvNodes responsibility to construct the filter. Only monitor the p2p messages if we have chosen to use bloom filters. 

Also we need to be able to support updating the filter, this may be out of scope for this PR but we need the ability update the filter without disconnecting the underlying peer.